In [ ]:

pip install docx2txt

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Created wheel for docx2txt: filename=docx2txt-0.8-py3-none-any.whl size=3980 sha256=6bcb0399131b4843c4551bd8466f57a84d3bea178532d2f99975daa5d8b09010
  Stored in directory: /root/.cache/pip/wheels/b7/20/b2/473e3aea9a0c0d3e7b2f7bd81d06d0794fec12752733d1f3a8
Successfully built docx2txt


In [ ]:
#import library
import docx2txt

In [ ]:
from  google.colab import files
uploaded = files.upload()

Saving Software Test Engineer Job Description.docx to Software Test Engineer Job Description.docx
Saving python-developer-resume-demo.docx to python-developer-resume-demo (2).docx


In [ ]:
#store the resume in variable
resume = docx2txt.process("python-developer-resume-demo (2).docx")
#print resume
print(resume)

ROBERT SMITH



Java/Python Developer





Phone: (0123)­456­789 | Email: info@qwikresume.com | Website: Qwikresume.com





SUMMARY



3+ years of SOFTWARE DEVELOPMENT EXPERIENCE in Python, Java, and IDEs ­ Sublime text, PyCharm, Eclipse, PyDev. Highly experienced in all phases of Software Development Life Cycle(SDLC). Expert in Web application development.





CORE COMPETENCIES



Technical : Frameworks Django, Web2py, Flask, Struts, Pylons And CSS Bootstrap, TurboGears, Web2py, Openstack Web Technologies HTML, CSS, DOM, SAX, Java Script, JQuery, AJAX, XML, Angular.JS Programming Languages Python, Java, SQL And PL/SQL. J2EE Technologies JSP, Java Servlets, JDBC And JMS. Version Control Git (GitHub)





PROFESSIONAL EXPERIENCE



Java/Python Developer



Computek Solutions ­ March 2015 – Present



Key Deliverables:



Written with object­oriented Python, Flask, SQL, Beautiful Soup, HTTPLIB2, JINJA2, HTML/CSS, BootStrap, Jquery, Linux, Sublime Text, GIT.

Developed the required XML 

In [ ]:
#store job description  into a veriable
job_description = docx2txt.process("Software Test Engineer Job Description.docx")
#print job description
print(job_description)

Experior Photonics, Inc.



750 Mitchell Road, Newbury Park, CA 91320



Ph: 805-499-3000



JOB DESCRIPTION



Title: Test Engineer – Fiber Optics (senior)







Reports To:



Technical Director



Targeted Start Date:



TBD



Work Hours:



Regular full time.



Roles and Responsibilities



Provides direction for the successful completion of development test programs, and to ensure compliance to contractual test requirements through development and support of processes, plans and procedures



	Generate test plans, procedures, and work instructions from customer requirements



	Setup and configure test setups, fixtures, and instrumentation as required by the test plans and or test standards



	Implement and deploy test fixtures, testing apparatus and machines and verify their appropriateness



	Ensure the accuracy and consistency of data results through thorough documentation processes



	Conduct and/or supervise testing by Test Technicians and other Test Engineers



	Opera

In [ ]:
#list of text
text = [resume, job_description]

In [ ]:
#import library to count the number of words
from sklearn.feature_extraction.text import CountVectorizer
cv = CountVectorizer()
count_matrix = cv.fit_transform(text)

In [ ]:
#import library for similar function
from sklearn.metrics.pairwise import cosine_similarity

#print hte similarity scores
print("\nSimilarity Scores:")
print(cosine_similarity(count_matrix))


Similarity Scores:
[[1.         0.52946852]
 [0.52946852 1.        ]]


similarity function shows that these two files have 0.529% similar

In [ ]:
#get the match percentage
matchpercentage = cosine_similarity(count_matrix)[0][1]*100
#converting value into 2 decimal
matchpercentage = round(matchpercentage, 2)
print("your resume match about "+ str(matchpercentage)+ "% of the job description")

your resume match about 52.95% of the job description
